oibackoff/index.d.ts

```diff
@@ -0,0 +1,46 @@
+// Type definitions for oibackoff
+// Project: https://github.com/chilts/oibackoff
+// Definitions by: Joshua DeVinney <https://github.com/geoffreak>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export declare function backoff(opts: BackoffOptions): BackoffInstance;
+
+export declare interface BackoffOptions {
+    maxTries?: number;
+    algorithm?: 'incremental' | 'exponential' | 'fibonacci';
+    delayRatio?: number;
+    maxDelay?: number;
+}
+
+export declare interface BackoffInstance {
+    (fn: Function, callback: Function): void;
```

Comment:
I've managed to create a working definition using generics, but it's over 1800 lines long (generated via a script). I'm working on producing a leaner version, but at least I know it is possible! [Version 1 script for reference](https://gist.github.com/geoffreak/f175a2c747f7b548c0b1a27e88eef283)